examples/plotting/great_arc_example.py

```diff
@@ -43,28 +42,18 @@
 plt.show()
 
 ###############################################################################
-# Now we can calculate the nearest integer pixels of the data that correspond
-# to the location of arc.
-
-pixels = np.asarray(np.rint(m.world_to_pixel(great_arc.coordinates())), dtype=int)
-x = pixels[0, :]
-y = pixels[1, :]
-
-###############################################################################
-# Get the intensity along the arc from the start to the end point.
-intensity_along_arc = m.data[y, x]
-
-###############################################################################
-# Define the angular location of each pixel along the arc from the start point
-# to the end.
-
-angles = great_arc.inner_angles().to(u.deg)
+# Now we can get the intensity along the great arc coordinates, along with the
+# angular distance from the start of the arc
+coords = great_arc.coordinates()
+intensity, intensity_coords = sunpy.map.extract_along_coord(m, coords)
+separation = intensity_coords.separation(intensity_coords[0]).to(u.arcsec)
 
 ###############################################################################
 # Plot the intensity along the arc from the start to the end point.
 
-plt.plot(angles, intensity_along_arc)
-plt.xlabel(f'Distance along arc [{angles.unit}]')
-plt.ylabel(f'Intensity [{m.unit}]')
+fig, ax = plt.subplots()
+ax.plot(separation, intensity)
+ax.set_xlabel(f'Separation from start of arc [{separation.unit}]')
+ax.set_ylabel(f'Intensity [{m.unit}]')
```

Comment:
Following our example gallery style guidelines for plotting: https://docs.sunpy.org/en/latest/dev_guide/contents/example_gallery.html#contribution-guidelines
```suggestion
plt.figure()
with quantity_support():
    plt.plot(separation, intensity)
```

Reply:
Our example gallery dev guidelines say that you shouldn't create an axes instance if you don't need to. I would say that adding axes labels does not necessitate the creation of an axes instance.

Reply:
I've opened https://github.com/sunpy/sunpy/pull/6822 to discuss this, and in the meantime removed the `Axes` object in this PR.